R/file_utils.R

```diff
@@ -608,15 +608,16 @@ load_twas_weights <- function(weight_db_files, conditions = NULL,
       }
       return(db)
     }))
-    # Check if region from all RDS files are the same
-    gene <- unique(names(all_data)[!names(all_data) %in% 'mnm_rs'])
+    
     # Combine the lists with the same region name
+    gene <- unique(names(all_data)[!names(all_data) %in% 'mnm_rs'])
```

Comment:
no fine-mapping results shouldbe loaded. The susie_weights_intermediate should be saved in the TWAS database directly as is.

Reply:
`mnm_rs` is a name for mnm results, so that it can be identified to pre-processed before merge with univariate twas weights 
mnm_rs does not contain any fine mapping result, below is a preview
```

$ mnm_rs         :List of 8
  ..$ Ast_mega_eQTL       :List of 6
  .. ..$ twas_weights      :List of 2
  .. .. ..$ mrmash_weights : Named num [1:6422] -1.83e-06 6.27e-06 4.45e-07 -2.73e-06 -4.02e-05 ...
  .. .. .. ..- attr(*, "names")= chr [1:6422] "chr11:84957209:G:C" "chr11:84957281:T:C" "chr11:84958954:T:C" "chr11:84959602:C:T" ...
  .. .. ..$ mvsusie_weights: Named num [1:6422] -4.87e-14 1.90e-13 -2.45e-14 -3.94e-14 -7.40e-13 ...
  .. .. .. ..- attr(*, "names")= chr [1:6422] "chr11:84957209:G:C" "chr11:84957281:T:C" "chr11:84958954:T:C" "chr11:84959602:C:T" ...
  .. ..$ twas_predictions  :List of 2
  .. .. ..$ mrmash_predicted : Named num [1:809] -0.0142 -0.0241 -0.1901 -0.1198 -0.0646 ...
  .. .. .. ..- attr(*, "names")= chr [1:809] "MAP15387421" "MAP26637867" "MAP34726040" "MAP46246604" ...
  .. .. ..$ mvsusie_predicted: Named num [1:809] -0.0714 -0.0946 -0.1968 -0.0946 -0.0714 ...
  .. .. .. ..- attr(*, "names")= chr [1:809] "MAP15387421" "MAP26637867" "MAP34726040" "MAP46246604" ...
  .. ..$ variant_names     : chr [1:6422] "chr11:84957209:G:C" "chr11:84957281:T:C" "chr11:84958954:T:C" "chr11:84959602:C:T" ...
  .. ..$ twas_cv_result    :List of 4
  .. .. ..$ sample_partition:'data.frame':	809 obs. of  2 variables:
  .. .. .. ..$ Sample: chr [1:809] "SM-CTDU8" "SM-CTDT7" "SM-CJFK5" "SM-CJEKR" ...
  .. .. .. ..$ Fold  : int [1:809] 1 1 1 1 1 1 1 1 1 1 ...
  .. .. ..$ prediction      :List of 2
  .. .. .. ..$ mrmash_predicted : num [1:809, 1] -0.0779 -0.101 -0.2089 -0.1019 -0.116 ...
  .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. ..$ : chr [1:809] "MAP15387421" "MAP26637867" "MAP34726040" "MAP46246604" ...
  .. .. .. .. .. ..$ : NULL
  .. .. .. ..$ mvsusie_predicted: num [1:809, 1] -0.0989 -0.1184 -0.2153 -0.1216 -0.1312 ...
  .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. ..$ : chr [1:809] "MAP15387421" "MAP26637867" "MAP34726040" "MAP46246604" ...
  .. .. .. .. .. ..$ : NULL
  .. .. ..$ performance     :List of 2
  .. .. .. ..$ mrmash_performance : num [1, 1:6] 1.70e-01 2.89e-02 2.76e-02 3.42e-06 3.93e-01 ...
  .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. ..$ : chr [1:6] "corr" "rsq" "adj_rsq" "pval" ...
  .. .. .. ..$ mvsusie_performance: num [1, 1:6] 1.94e-01 3.75e-02 3.62e-02 1.15e-07 3.99e-01 ...
  .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. ..$ : chr [1:6] "corr" "rsq" "adj_rsq" "pval" ...
  .. .. ..$ time_elapsed    : 'proc_time' Named num [1:5] 287.6 49 9066.2 6203.8 21.9
  .. .. .. ..- attr(*, "names")= chr [1:5] "user.self" "sys.self" "elapsed" "user.child" ...
  .. ..$ total_time_elapsed: 'proc_time' Named num [1:5] 288.1 49.2 9067.3 6203.8 22.1
  .. .. ..- attr(*, "names")= chr [1:5] "user.self" "sys.self" "elapsed" "user.child" ...
  .. ..$ region_info       :List of 3
  .. .. ..$ region_coord:'data.frame':	1 obs. of  3 variables:
  .. .. .. ..$ chrom: chr "11"
  .. .. .. ..$ start: int 86069881
  .. .. .. ..$ end  : int 86069882
  .. .. ..$ grange      :'data.frame':	1 obs. of  3 variables:
  .. .. .. ..$ chrom: chr "11"
  .. .. .. ..$ start: int 84957175
  .. .. .. ..$ end  : int 87360000
  .. .. ..$ region_name : chr [1:9] "ENSG00000073921" "ENSG00000073921" "chr11_ENSG00000073921_Q13492" "ENSG00000073921" ...
  ..$ DLPFC_Bennett_pQTL  :List of 6
```

Reply:
I see, that makes sense.